samples/cpp/benchmark_app/utils.hpp

```diff
@@ -149,3 +154,122 @@ void convert_io_names_in_map(
     }
     map = new_map;
 }
+
+void verifyBinaryFile(std::ifstream& binaryFile, const std::string& fileName, const unsigned long inputSize);
+template <typename T>
+void verifyNumpyFile(std::ifstream& binaryFile,
+                     const std::string& fileName,
+                     const ov::Shape& inputShape,
+                     const unsigned long inputSize) {
+    auto fullFileSize = static_cast<std::size_t>(binaryFile.tellg());
+    binaryFile.seekg(0, std::ios_base::beg);
+    OPENVINO_ASSERT(binaryFile.good(), "Can not read ", fileName);
+
+    std::string magic_string(6, ' ');
+    binaryFile.read(&magic_string[0], magic_string.size());
+    OPENVINO_ASSERT(magic_string == "\x93NUMPY",
+                    "Numpy file has incorrect magic string. File ",
+                    fileName,
+                    " might be corrupted");
+
+    binaryFile.ignore(2);  // Skip major and minor version bytes
+    unsigned short headerSize;
+    binaryFile.read((char*)&headerSize, sizeof(headerSize));
+
+    std::string header(headerSize, ' ');
+    binaryFile.read(&header[0], header.size());
+
+    int idx, from, to;
+
+    // Verify fortran order is false
+    const std::string fortranKey = "'fortran_order':";
+    idx = header.find(fortranKey);
+    OPENVINO_ASSERT(idx != -1, "Numpy file is missing fortran_order key. File ", fileName, " might be corrupted");
+    from = header.find_last_of(' ', idx + fortranKey.size()) + 1;
+    to = header.find(',', from);
+    auto fortranValue = header.substr(from, to - from);
+    OPENVINO_ASSERT(fortranValue == "False", "File ", fileName, " was saved in Fortran order, which is not supported");
+
+    // Verify array shape matches the input's
+    const std::string shapeKey = "'shape':";
+    idx = header.find(shapeKey);
+    OPENVINO_ASSERT(idx != -1, "Numpy file is missing shape key. File ", fileName, " might be corrupted");
+
+    from = header.find('(', idx + shapeKey.size()) + 1;
+    to = header.find(')', from);
+
+    std::string shapeStr = header.substr(from, to - from);
+    std::vector<size_t> numpyShape;
+
+    if (!shapeStr.empty()) {
```

Comment:
Should you throw if it is empty?

Reply:
I am not sure if this is the case, but I believe openvino allows for no-shape tensors (they would be basically scalars), so I maybe should add `else { numpyShape.push_back(0); }`. I will test this and let you know if it works

Reply:
Changed the logic, no longer applies